### PR TITLE
Fixing some bugs so the Dataflow GA version bump doesn't break us.

### PR DESCRIPTION
**Pull request by davidadamsphd**

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/DataflowReadsPipelineTest.java [hunk 1]
```diff
@@ -24,6 +24,8 @@
 import java.util.Collections;
 import java.util.List;
 
+import static org.broadinstitute.hellbender.transformers.ReadTransformer.*;
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/DataflowReadsPipelineTest.java]: Why use this static import?  We typically avoid them.

File: src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorWorker.java [hunk 2]
```diff
@@ -242,8 +242,9 @@ private ReadTransformer makeReadTransform() {
      * The "knownLocations" list doesn't need to be complete but it must includes all those that overlap with the read.
      */
     public void apply(GATKRead originalRead, final ReferenceContext ref, List<? extends Locatable> knownLocations) {
+        final GATKRead copy = originalRead.copy();  // REMOVE ME?
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorWorker.java]: What's with the remove me?

File: src/test/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/DataflowReadsPipelineTest.java (hunk 1 above)
Comment (davidadamsphd) [inline src/test/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/DataflowReadsPipelineTest.java]: IntelliJ thought it was needed (wouldn't compile without it).

File: src/main/java/org/broadinstitute/hellbender/utils/read/GoogleGenomicsReadToGATKReadAdapter.java [hunk 3]
```diff
@@ -60,6 +60,11 @@ public GoogleGenomicsReadToGATKReadAdapter( final Read genomicsRead ) {
         private static final long serialVersionUID = 1l;
 
         @Override
+        public String getEncodingId() {
+            return "GoogleGenomicsReadToGATKReadAdapterCoder";
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/utils/read/GoogleGenomicsReadToGATKReadAdapter.java]: When do we need getEncodingId?

File: src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorWorker.java (hunk 2 above)
Comment (davidadamsphd) [inline src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorWorker.java]: Opps! It was there to help me keep track of this that _might_ be the cause. It  was the cause (so I'm removing the comment).

File: src/test/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/DataflowReadsPipelineTest.java (hunk 1 above)
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/DataflowReadsPipelineTest.java]: Wouldn't compile?  Or showed a red line? There's an intellij bug with their evaluation of this line but it should compile fine despite the warning.

**Comment by lbergelson** [comment]
What test did toString() break.  That seems like a bad test.

**Comment by davidadamsphd** [comment]
It's (most likely) a bad test:
GATKReadAdaptersUnitTest.testGetAndSetBases[2], I think it's getting a null SamRecord (which we don't check for).

We should fix the test and not allow null sams... but not in this PR :)

**Comment by lbergelson** [comment]
@davidadamsphd Looks good.  Just that static import.  I think it should probably change back unless it's actually a compile error.   Intelllij incorrectly marks it though.  There's a fix coming in some future version.

**Comment by lbergelson** [comment]
What test was depending on toString?  We should maybe update that test...

**Comment by droazen** [comment]
That test (GATKReadAdaptersUnitTest.testGetAndSetBases) is not a bad test -- it is a unit test for the `getBases()` and `getBasesString()` methods in the `GATKRead` interface. Its test cases include reads with no bases (which is allowed).

**Comment by lbergelson** [comment]
Ah, it's because of https://github.com/samtools/htsjdk/issues/297.  But why is the test calling toString()?

**Comment by droazen** [comment]
It's not calling `toString()` -- it only calls `getBasesString()` (since it is unit-testing that method).

**Comment by lbergelson** [comment]
Why is it failing when you define toString then?  It must be crashing somewhere in the test suit itself.

**Comment by droazen** [comment]
Possibly TestNG is calling `toString()` on the test method parameters when creating its report of test cases that passed / failed, and this triggers the htsjdk bug you linked to?

**Comment by lbergelson** [comment]
:+1: Merge when ready.

**Pull request merged**